### ISSUE TITLE
As a logged in and authorized user, I should be able to access and edit audits

### ISSUE THREAD

**Issue by petermarks-gsa** [GSA-TTS/FAC#332]
## User story
As a logged in and authorized user, I should be able to access and edit audits

## High-level solution
Implement audit level permissions: Some sort of way to verify the user has access to the audit they're trying to access via url validation

## Acceptance criteria
* [ ] When I try to access an audit I have access to and am logged in, I should be able to navigate to and view/edit that audit.
* [ ] When I try to access an audit I do not have access to and am logged in, I should be redirected to the home page with alert/flash. 
* [ ] When I try to access an audit I have access to and am **NOT** logged in, I should be redirected to the home page with alert/flash. 

## Tasks
*   Backend endpoint #398 covers the following two tasks previously captured here:
    * 🅛 Determine and implement URL routing for individual submissions (report id?)
    * 🅜 Back end - implement access denial
*   Frontend side of #398 is #399.
* #401
* ~~Placeholder audit start page~~
* (Design/copy) - Copy for access denial in USWDS Alert/notification component

## Related, but separate stories
* https://github.com/GSA-TTS/FAC/issues/359
* https://github.com/GSA-TTS/FAC/issues/319
* https://github.com/GSA-TTS/FAC/issues/212
* https://github.com/GSA-TTS/FAC/issues/342
* https://github.com/GSA-TTS/FAC/issues/343
* https://github.com/GSA-TTS/FAC/issues/388
* https://github.com/GSA-TTS/FAC/issues/486

**Comment by petermarks-gsa** [comment]
@sreuber When you get a chance, please send along an updated design for the "My audit submissions" page

**Comment by tadhg-ohiggins** [comment]
The third AC doesn't read right to me; currently it says that if trying to go to an audit I have access to when I am logged in should redirect to home page. Should this be addressing what happens if I'm not logged in?

**Comment by petermarks-gsa** [comment]
@tadhg-ohiggins Good catch, I added a **NOT** in there: 

* [ ] When I try to access an audit I have access to and am **NOT** logged in, I should be redirected to the home page with alert/flash.

**Comment by petermarks-gsa** [comment]
@tadhg-ohiggins Are we good to remove "needs scoping" now?

**Comment by heymatthenry** [comment]
@petermarks-gsa @tadhg-ohiggins I'm calling this `XL` because it has multiple constituent tickets, most of which are themselves `L`s.

**Comment by heymatthenry** [comment]
@petermarks-gsa Should the first AC on this be:

> When I try to access an audit I have access to and am logged in, I should be able to navigate to and view/edit that audit.

Or am I missing something?

**Comment by petermarks-gsa** [comment]
@heymatthenry Hmm, yeah somehow that got changed. I changed it to what you have. Thanks!

**Comment by tadhg-ohiggins** [comment]
@heymatthenry @petermarks-gsa I believe that currently we don't have the functionality for passing the original destination along through the login flow.

While that's how it should work in the long run, we should consider that an enhancement (it's captured [here](https://github.com/GSA-TTS/FAC/issues/407)).

**Comment by petermarks-gsa** [comment]
@tadhg-ohiggins Agree routing to original destination is out of scope for this. Thanks for making a breakout.

**Comment by petermarks-gsa** [comment]
Per discussion today, sounds like this will be done once https://github.com/GSA-TTS/FAC/issues/391 is done. Will verify with runthrough.

**Comment by tadhg-ohiggins** [comment]
This is also dependent on #399

**Comment by tadhg-ohiggins** [comment]
@tadhg-ohiggins to verify that completion of the other issues means this is done (or identify remaining issues required).

**Comment by pkigongo** [comment]
Blocked by general information form. Blocked until #563 is done

**Comment by mogul** [comment]
Now fully unblocked. We noted that auditors and auditees don't see submissions that they didn't create in the list, so we need to fix that!
